internal/agent/client/compose.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+)
+
+const defaultPodmanTimeout = time.Minute
```

Comment:
Will one minute be enough to bring up applications, especially if they require image downloads?

Reply:
all transient errors should be retried

Reply:
I am open to longer timeouts but if we retry and only pull what was not cached this feels reasonable. It certainly could be proved wrong. I like shorter timeouts and faster feedback via logs/status. But if you have a number in mind let me know.